Source/Initialization/WarpXInitData.cpp

```diff
@@ -1215,3 +1269,152 @@ void WarpX::CheckKnownIssues()
                 ablastr::warn_manager::WarnPriority::low);
         }
 }
+
+#if defined(WARPX_USE_OPENPMD) && !defined(WARPX_DIM_1D_Z)
+void
+WarpX::ReadExternalFieldsFromFile (std::string read_fields_from_path, MultiFab* mf,
+std::string F_name, std::string F_component)
+{
+    // Get WarpX domain info
+    auto& warpx = WarpX::GetInstance();
+    amrex::Geometry const& geom0 = warpx.Geom(0);
+    const amrex::RealBox& real_box = geom0.ProbDomain();
+    const auto dx = geom0.CellSizeArray();
+    amrex::IntVect nodal_flag = mf->ixType().toIntVect();
+
+    // Loop over boxes
+    for (MFIter mfi(*mf, TilingIfNotGPU()); mfi.isValid(); ++mfi)
+    {
+        // Read external field openPMD data
+        auto series = openPMD::Series(read_fields_from_path, openPMD::Access::READ_ONLY);
+        auto iseries = series.iterations.begin()->second;
+        auto F = iseries.meshes[F_name];
+        auto offset = F.gridGlobalOffset();
+        auto offset0 = offset[0];
+        auto offset1 = offset[1];
+#if defined(WARPX_DIM_3D)
+        auto offset2 = offset[2];
+#endif
+        auto d = F.gridSpacing<long double>();
+        auto d0 = d[0];
+        auto d1 = d[1];
+#if defined(WARPX_DIM_3D)
+        auto d2 = d[2];
+#endif
+
+        auto FC = F[F_component];
+        auto extent = FC.getExtent();
+
+        auto box = mfi.growntilebox();
+
+        // Determine the chunk data that will be loaded.
+        // Now, the full range of data is loaded.
+        // Loading chunk data can speed up the process.
+        // Thus, `chunk_offset` and `chunk_extent` should be modified accordingly in another PR.
+        openPMD::Offset chunk_offset = {0,0,0};
+        openPMD::Extent chunk_extent = {extent[0],extent[1],extent[2]};
+
+        auto FC_chunk_data = FC.loadChunk<double>(chunk_offset,chunk_extent);
+        series.flush();
+        auto FC_data_host = FC_chunk_data.get();
+
```

Comment:
Add assertions about axis labels and data order matching expectation below, or generalize to respect them